### DISCUSSION TITLE
Question about stickycookies and how MITM treats cookies in general.

### DISCUSSION THREAD

**Discussion by creamtart** [mitmproxy/mitmproxy#6342] [General]
Hello. Basically I'm using MITM in reverse proxy mode, and for the site to work properly it needs to get the set-cookie value and set it as a cookie. This only works when I use stickycookies but from what I understand this sets the cookies for the entire MITM session, meaning if one user accesses from one browser, the cookie is set for all. Is there a way to make it use the set-cookie value per user session, so that when a user closes the browser and disconnects from another IP the cookies are reset? Thanks to everyone in advance.

**Comment by Prinzhorn** [comment]
> This only works when I use stickycookies

I don't think it should, if you can provide clear steps to reproduce then this would be a bug. Headers, including `set-cookie`, are supposed to be proxied.

```sh
mitmproxy --mode reverse:https://google.com
```

Set-Cookie is present:

```
$ curl --include http://localhost:8080
HTTP/1.1 301 Moved Permanently
Location: https://www.google.com/
Content-Type: text/html; charset=UTF-8
Content-Security-Policy-Report-Only: object-src 'none';base-uri 'self';script-src 'nonce-1cR7RfhiGcx8RhcQ77dDAg' 'strict-dynamic' 'report-sample' 'unsafe-eval' 'unsafe-inline' https: http:;report-uri https://csp.withgoogle.com/csp/gws/other-hp
Date: Fri, 01 Sep 2023 13:29:18 GMT
Expires: Fri, 01 Sep 2023 13:29:18 GMT
Cache-Control: private, max-age=2592000
Server: gws
Content-Length: 220
X-XSS-Protection: 0
X-Frame-Options: SAMEORIGIN
Set-Cookie: CONSENT=PENDING+618; expires=Sun, 31-Aug-2025 13:29:18 GMT; path=/; domain=.google.com; Secure
P3P: CP="This is not a P3P policy! See g.co/p3phelp for more info."
Alt-Svc: h3=":443"; ma=2592000,h3-29=":443"; ma=2592000

<HTML><HEAD><meta http-equiv="content-type" content="text/html;charset=utf-8">
<TITLE>301 Moved</TITLE></HEAD><BODY>
<H1>301 Moved</H1>
The document has moved
<A HREF="https://www.google.com/">here</A>.
</BODY></HTML>

```

**Reply by Prinzhorn** [reply]
I think the problem you're facing might be the mismatch with the `domain` part of the cookie and the browser refusing to actually set it. You can rewrite them using an addon if it doesn't work for your setup otherwise (not all cookies set `domain`).
